controller/src/main/java/org/jboss/as/controller/RunningModeControl.java

```diff
@@ -17,6 +17,7 @@ public class RunningModeControl {
     private volatile boolean useCurrentConfig;
     private volatile String newBootFileName;
     private volatile Boolean suspend;
+    private volatile boolean applyConfigurationExtension;
```

Comment:
RunningModeControl has evolved keeping more stuff than only the required information to provide control over the server's current RunningMode. However, instead of adding information on whether the configuration extensions should be applied, wouldn't be easier if we just save here the value of extensibleConfigurationPersister.hasStored()?

With this information and the other field values we could derive in  `ConfigurationExtension.shouldProcessOperations()` whether the yaml configuration should be applied again after a reload.

Reply:
I don't really understand  what you are proposing there.

Reply:
@ehsavoie I mean, move all the logic that calculates the `applyConfigurationExtension` from the reload handler to `ConfigExtension.shouldProcessOperations()` method and instead save in the RunningModeControl the flag that indicates the persistent storage has been stored. e.g extensibleConfigurationPersister.hasStored() value.

With the RunningModeControl class field values you should be able to calculate on `ConfigExtension.shouldProcessOperations()` whether the YAML operations should be loaded again.

Understanding the logic and purpose of `applyConfigurationExtension` calculation by looking at the handlers is a bit unclear, if you prefer to keep what you currently have, at least write a java doc explaining the applyConfigurationExtension purpose and meaning on RunningModeControl accessors.

Out of curiosity, I am not sure how the stored method is invoked even when we are using a read-only configuration file policy, but, if the user writes something after reloading, all the configuration, including those derived from the YAML files, will be available from the persistent storage? I understand that's the reason why you don't want to apply the YAML config again to do not overwrite any modification done after the previous reloading

Reply:
When we are in read-only mode the changes are persisted to the standalone-boot.xml so that when you reload you keep them. That's when the store method is invoked.
When you start or restart the standalone-boot.xml is overwritten by the standalone.xml.